src/components/EventCard.astro

```diff
@@ -0,0 +1,191 @@
+---
+import Icon from "./Icon.astro";
+
+interface Props {
+  title: string;
+  description: string;
+  date: string;
+  time: string;
+  location: string;
+  poster_img: string;
+  link: string;
+}
+
+const { title, description, date, time, poster_img, location, link } = Astro.props as Props;
+---
+
+<a class="card" href={link}>
+  <div class="card-img-wrapper">
+    <img class="card-img" src={poster_img} alt="Workshop Poster" />
+  </div>
+  <div class="card-content">
+    <div class="card-header">
+      <h3 class="card-title">{title}</h3>
+      <p class="card-desc">{description}</p>
+    </div>
+    <div class="card-footer">
+      <div class="row">
+        <Icon icon="calendar" />
+        {date}
+      </div>
+      <div class="row">
+        <Icon icon="clock" />
+        {time}
+      </div>
+      <div class="row">
+        <Icon icon="location" />
+        {location}
+      </div>
+    </div>
+  </div>
+
+  <div class="reveal-on-hover">
+    <p>Click to learn more &gt;&gt;</p>
+  </div>
+</a>
+
+<style lang="scss">
+  @import "../styles/index.scss";
+
+  .card {
+    color: $text-primary;
+
+    position: relative;
+    flex-direction: column;
+
+    display: flex;
+    align-items: center;
+
+    border-radius: 12px;
+
+    @include raised;
+
+    width: 28%;
+    transition: all 300ms;
```

Comment:
Add this to `.card-img` and `.reveal-on-hover` since they are also changing on hover.